homeassistant/components/efergy/sensor.py

```diff
@@ -59,11 +60,14 @@
     CONF_BUDGET: SensorEntityDescription(
         key=CONF_BUDGET,
         name="Energy Budget",
+        device_class=DEVICE_CLASS_MONETARY,
+        native_unit_of_measurement=CURRENCY_DOLLAR,
```

Comment:
This is not an ISO standard currency code.

https://en.m.wikipedia.org/wiki/ISO_4217